firebase.agent.nut

```diff
@@ -383,8 +383,42 @@ class Firebase {
         }.bindenv(this);
     }
 
+    function _isValidJson(text) {
+        try {
+            http.jsondecode(text);
+        } catch (err) { 
+            return false; 
+        }
+        return true; 
+    }
+
     // parses event messages
-    function _parseEventMessage(text) {
+    function _parseEventMessage(input) {
+        local text = _bufferedInput + input;
+        local dataIndex = text.find("data");
+        local bracketIndex = text.find("{");
+
+        if (dataIndex && dataIndex < bracketIndex) {
+            if (!_isValidJson(text.slice(bracketIndex))) { 
+                _bufferedInput = text;
+                return []; 
+            }
+        } else {
+            if (bracketIndex == null && text.find("null")) {
+                _bufferedInput = text;   
```

Comment:
There is something wrong here. You should either return right after this line or it doesn't make any sense to me.